src/javascript/class-scanner.ts

```diff
@@ -335,6 +347,221 @@ interface CustomElementDefinition {
   definition?: ElementDefineCall;
 }
 
+class PrototypeMemberFinder implements Visitor {
+  readonly members = new Map<string,
+    {methods: Map<string, ScannedMethod>, properties: Map<string, ScannedProperty>}>();
+  private readonly _document: JavaScriptDocument;
+
+  constructor(document: JavaScriptDocument) {
+    this._document = document;
+  }
+
+  enterExpressionStatement(node: babel.ExpressionStatement) {
+    if (babel.isAssignmentExpression(node.expression)) {
```

Comment:
If you pull `node.expression` out into a local const then `babel.isAssignmentExpression` and friends will narrow its type, so you can avoid checking the type again inside your `_createMemberFrom...` methods

Reply:
i've instead made the methods take what they say they take. it was a bit confusing before, being called `createMemberFromAssignment` but not taking an `AssignmentExpression`... so now we just pass the expression in.